eth/api.go

```diff
@@ -644,16 +683,15 @@ func (s *PublicBlockChainAPI) NewBlocks(ctx context.Context, args NewBlocksArgs)
 
 	// add a callback that is called on chain events which will format the block and notify the client
 	s.muNewBlockSubscriptions.Lock()
-	s.newBlockSubscriptions[subscription.ID()] = func(e core.ChainEvent) error {
-		if notification, err := s.rpcOutputBlock(e.Block, args.IncludeTransactions, args.TransactionDetails); err == nil {
+	s.newBlockSubscriptions[subscription.ID()] = func(e core.ChainEvent) (err error) {
```

Comment:
Please don't use named returns to avoid declaring a local variable, especially when it's just used as a local variable and not even returned. The cleanest version of this code would be:

```
notification, err := s.rpcOutputBlock(e.Block, args.IncludeTransactions, args.TransactionDetails)
if err == nil {
	return subscription.Notify(notification)
}
glog.V(logger.Warn).Info("unable to format block %v\n", err)		
return nil
```